src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -154,72 +170,157 @@ export default class SimidMapCreative extends BaseSimidCreative {
+    };
+    const service = new google.maps.places.PlacesService(this.map_);
+    service.nearbySearch(request, this.displayResults_.bind(this));
+    this.createTravelChoices_();
+  }
 
-/**
- * Displays the closest business locations to a user's current location.
- * @param {!Object} results An array of Place Results from the search query.
- * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
- *  by the PlacesService on the completion of its searches.
- * @private 
- */
-displayResults_(results, status) {
+  /**
+   * Displays the closest advertisement's locations to a user's current location.
+   * @param {!Object} results An array of Place Results from the search query.
+   * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
+   *  by the PlacesService on the completion of its searches.
+   * @private 
+   */
+  displayResults_(results, status) {
     if (status == google.maps.places.PlacesServiceStatus.OK) {
+      this.activeLocation_ = results[0].geometry.location;
       for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
         this.placeMapMarker_(results[i]);
       }
+      this.displayDirections_();
+    } else {
+      const statusErrorMessage = {
+        message: "ERROR: Failed to complete search: "+status,
+      };
+      this.simidProtocol.sendMessage(CreativeMessage.LOG, statusErrorMessage);
+      this.playAd_();
     }
-}
+  }
 
-/**
- * Creates and displays a marker on the map representing a given place.
- * @param {!Object} place A Place Result object.
- * @private 
- */
-placeMapMarker_(place) {
-  const placeIcon = {
-    url: this.markerImage_,
-    scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
-  };
-  const placeMarker = new google.maps.Marker({
-    map: this.map_,
-    position: place.geometry.location,
-    icon: placeIcon
-  });
-}
+  /**
+   * Creates and displays a marker on the map representing a given place.
+   * @param {!Object} place A Place Result object.
+   * @private 
+   */
+  placeMapMarker_(place) {
+    const placeIcon = {
+      url: this.markerImage_,
+      scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
+    };
+    const placeMarker = new google.maps.Marker({
+      map: this.map_,
+      position: place.geometry.location,
+      icon: placeIcon
+    });
+    ///Recalculate directions if a different active marker is selected.
+    placeMarker.addListener('click', () => {
+      this.activeLocation_ = place.geometry.location;
+      this.displayDirections_();
+    });
+  }
+
+  /**
+   * Displays the directions between the user's current location and current active location.
+   * @private 
+   */
+  displayDirections_() {
+    this.directionsRenderer_.setMap(this.map_);
+    this.calculateRoute_();
+  }
+
+  /**
+   * Creates a drop down menu where users can choose between
+   * different modes of travel to display directions for.
+   * @private 
+   */
+  createTravelChoices_() {
```

Comment:
It just feels like a lot of this stuff should come packaged into the maps API and we shouldn't have to directly specify it.  I mean, what options and choices to we get if we just use the default google maps?

Reply:
What do you mean by the default google maps? The transport options I have listed here are all of the options offered by the Google Maps API.